### DISCUSSION TITLE
Plot.geo highlight country on hover: border thickness varies on swapping neighbouring countries

### DISCUSSION THREAD

**Discussion by espinielli** [observablehq/plot#2260] [Q&A] [answered]
I tried to highlight countries on hover but I see overlapping borders of different thickness when switching from one to its neighborough.
For example in [this notebook](https://observablehq.com/d/d423284aa770eb88)
1. Hover over Russia: the border with Mongolia is thin
   <img width="422" alt="Screenshot 2024-12-17 at 09 08 01" src="https://github.com/user-attachments/assets/78cf7231-963c-49f1-a5e7-981360e59313" />

1. Hover hover Mongolia: the border with Russia is now thicker
   <img width="400" alt="Screenshot 2024-12-17 at 09 08 27" src="https://github.com/user-attachments/assets/7edd67d3-9813-429d-9986-ba74c6535309" />

**Q**: How do you get constant/consistent thickness?
**Q**: Also is the method (trick?) I used the right way to highlight polygons in Plot.geo?
```js
Plot.plot({
  projection: "equal-earth",
  width,
  marks: [
    Plot.graticule(),
    Plot.geo(land, { fill: "#AAA" }),
    Plot.geo(countries, {
      // give it a class for easier CSS matching when highlighting
      className: "country",
      fill: (d) => +d.id,
      title: (d) => d.properties.name
    }),
    Plot.sphere(),
    // highlight
    () =>
      svg`<style>
        .country path:hover {stroke: red; stroke-width: 2; stroke-opacity: 1.0; transition: stroke-opacity .1s }
      `
  ]
})
```

Any helps or suggestions are welcome.
Thanks

**Comment by Fil** [comment, marked as answer]
It depends on the order in which the shapes are drawn (_ie._ Mongolia is drawn after Russia, so its _fill_ is drawn on top of Russia's _stroke_).

To fix this you can create two marks (one for the fill, one for the stroke); use javascript (e.g. d3.selection's [raise](https://d3js.org/d3-selection/modifying#selection_raise)); or clip the shapes by themselves  to ensure the stroke is drawn “inwards” (hard to do with Plot). 

I've sent you a suggestion that implements the first suggestion
https://observablehq.com/compare/d423284aa770eb88...23b3a136ce8e492a

**Reply by espinielli** [reply]
Thanks I applied it to https://observablehq.com/@espinielli/world-firs-with-zoom-and-rotate and it works like a charm!
